lib/orbit-common/cache.js

```diff
@@ -670,10 +670,7 @@ var Cache = Class.extend({
     if (this.retrieve([type, id])) {
       var op = this._addLinkOp(type, id, key, value);
 
-      // Apply operation only if necessary
-      if (!this.retrieve(op.path)) {
-        this.transform(parentOperation.spawn(op));
-      }
+      this.transform(parentOperation.spawn(op));
```

Comment:
This old if here was causing it to not properly "overwrite" if some link were to cause the inverse hasOne to be "replaced"